File: src/main/java/WaterBottle.java
public class WaterBottle {
    private double volume;
    private double startingVolume;

    public WaterBottle(double volume, double startingVolume){
        this.volume = volume;
        this.startingVolume = startingVolume;
    }

    public double getVolume(){
        return this.volume;
    }

    public void drink(double amount){
        this.volume -= amount;
    }

    public void empty(){
        this.volume = 0;
    }

    public void fill(){
        this.volume = this.startingVolume;
    }

}

File: src/main/java/WaterBottle.kt
class WaterBottle(volume: Double, private val startingVolume: Double) {
    var volume: Double = 0.toDouble()
        private set

    init {
        this.volume = volume
    }

    fun drink(amount: Double) {
        this.volume -= amount
    }

    fun empty() {
        this.volume = 0.0
    }

    fun fill() {
        this.volume = this.startingVolume
    }


}

